### DISCUSSION TITLE
Pass the id of a button to a callback function

### DISCUSSION THREAD

**Discussion by AliE89** [Kitware/trame#373] [Q&A] [answered]
Good evening all,

I am new in the wonderful world of Trame! 

It would be great if somebody can suggest me a way to pass the id of the button to his binded function. 

Here is my sample code:

                   variables=['a','b','c']

                    def my_fun(event, id):
                          print(event, id)

                    vuetify.VSelect(
                        id='my_button',
                        label="Contour by",
                        v_model=(f"v_model_{pname}", 0),
                        items=("array_list", variables),
                        hide_details=True,
                        dense=True,
                        outlined=True,
                        change=(my_fun, '[$event, ??? ]')
                    )

I would like to pass both the event and the id of the button to my_fun as soon as I change that button in the GUI.  Any particular trick to do this?

Thanks a lot for the help!

**Comment by jourdain** [comment, marked as answer]
Thanks for your kind words. 

The simplest way would be to do the following

```python
vuetify.VSelect(
    # ...
    change=(my_fun, '[$event, 'my_button']')
)
```

If the $event was an actual JS event with the HTML target element you could do something like `$event.target.id` but that is not the case with VSelect/@change.

HTH

**Reply by AliE89** [reply]
Hey Jourdain,  thank you. Could not believe to my eyes when I met this tasty stuff! 

Your suggestion is indeed good, actually I am creating buttons in a for loop and my_button is a string created on the fly during the loop. Doing this works ok:
    change=(update_contour_by, f"[$event, '{my_button}']")

I missed to include the single quotes '', rookie mistake :) 

Cheers
